src/Optimizely/Event/Builder/EventBuilder.php

```diff
@@ -60,147 +59,174 @@ class EventBuilder
     ];
 
     /**
-     * @var array Associative array of parameters to be sent for the event.
-     */
-    private $_eventParams;
-
-    /**
-     * Helper function to reset event params.
-     */
-    private function resetParams()
-    {
-        $this->_eventParams = [];
-    }
-
-    /**
-     * @return array Params for the event.
-     */
-    private function getParams()
-    {
-        return $this->_eventParams;
-    }
-
-    /**
-     * Helper function to set parameters common to impression and conversion event.
+     * Helper function to get parameters common to impression and conversion event.
      *
      * @param $config ProjectConfig Configuration for the project.
      * @param $userId string ID of user.
      * @param $attributes array Attributes of the user.
      */
-    private function setCommonParams($config, $userId, $attributes)
+    private function getCommonParams($config, $userId, $attributes)
     {
-        $this->_eventParams[PROJECT_ID] = $config->getProjectId();
-        $this->_eventParams[ACCOUNT_ID] = $config->getAccountId();
-        $this->_eventParams[REVISION] = $config->getRevision();
-        $this->_eventParams[VISITOR_ID] = $userId;
-        $this->_eventParams[CLIENT_ENGINE] = self::SDK_TYPE;
-        $this->_eventParams[CLIENT_VERSION] = self::SDK_VERSION;
-        $this->_eventParams[USER_FEATURES] = [];
-        $this->_eventParams[IS_GLOBAL_HOLDBACK] = false;
-        $this->_eventParams[TIME] = time()*1000;
-        if (!isset($attributes)) {
-            $attributes = [];
-        }
+        $visitor = [
+            'snapshots'=> [],
+            'visitor_id'=> $userId,
+            'attributes' => []
+        ];
 
-        forEach ($attributes as $attributeKey => $attributeValue) {
+        $commonParams = [
```

Comment:
You don't need to change the style, please set in the EventParams.

Reply:
There is no object-level _eventParams anymore. This method populates common parameters for both events and returns the array. The array name does not make a difference. I have followed the [nodeJsSDK](https://github.com/optimizely/node-sdk/blob/master/lib/core/event_builder/index.js) pattern